reflectapi-demo/clients/rust/generated/src/generated.rs

```diff
@@ -49,8 +49,9 @@ pub mod interface {
         ) -> Result<reflectapi::Empty, reflectapi::rt::Error<reflectapi::Empty, C::Error>> {
             reflectapi::rt::__request_impl(
                 &self.client,
-                &self.base_url,
-                "/health.check",
+                self.base_url
```

Comment:
van base_url cause the expect to fail?

Reply:
It can. I can add an `assert!(!base_url.cannot_be_a_base())` in `Interface::new()`. That should catch most causes of panics early. I still it's fine to panic since it's a user error to pass a bad url that would cause `join` to fail.

```
        if let Some(base_url) = self.base_url {
            if input.starts_with('#') {
                self.fragment_only(base_url, input)
            } else if base_url.cannot_be_a_base() {
                Err(ParseError::RelativeUrlWithCannotBeABaseBase)
            } else {
                let scheme_type = SchemeType::from(base_url.scheme());
                if scheme_type.is_file() {
                    self.parse_file(input, scheme_type, Some(base_url))
                } else {
                    self.parse_relative(input, scheme_type, base_url)
                }
            }
        } else {
            Err(ParseError::RelativeUrlWithoutBase)
        }

```

Reply:
I do not like panicking or asserting in general, but especially for user input "it's fine to panic since it's a user error to pass a bad url"... 

Previously we submitted that to the networking layer (reqwest) and it handled invalid input responding with can not connect or something. Not sure why we needed this change... If we keep it the only way to handle it is to validate the base url at the place where it is supplied, meaning Interface::new() would need to become Interface::try_new() ?

Reply:
having url used instead of string would still have that problem as trailing slashes and double slashes are allowed in url.

Reply:
yes, but `"localhost/".join("/api")` is handled correctly. `format("{}{}", "localhost/", "/api")` results in double slash.

Reply:
More generally I think it's just using the right type for the job. I usually avoid string for everything except things where it's actually an arbitrary string such as a `description` because there are basically no restrictions.

Reply:
"I would much rather code panic early on error than fail in strange ways like this." - ok, then maybe the best is to assert and return an error when the base url is provided, i.e. from Interface::try_new perhaps... and assert in the handlers, but that failing assert would fire because of our bug not because of user's input

Reply:
changed to `try_new` and will panic in handlers.